hazelcast/src/main/java/com/hazelcast/collection/impl/queue/operations/QueueMergeBackupOperation.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.collection.impl.queue.operations;
+
+import com.hazelcast.collection.impl.queue.QueueContainer;
+import com.hazelcast.collection.impl.queue.QueueDataSerializerHook;
+import com.hazelcast.collection.impl.queue.QueueItem;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.spi.impl.MutatingOperation;
+import com.hazelcast.spi.merge.SplitBrainMergePolicy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Contains multiple backup QueueItems {@link SplitBrainMergePolicy}.
+ *
+ * @since 3.10
+ */
+public class QueueMergeBackupOperation extends QueueOperation implements MutatingOperation {
+
+    private Collection<QueueItem> backupItems;
+
+    public QueueMergeBackupOperation() {
+    }
+
+    public QueueMergeBackupOperation(String name, Collection<QueueItem> backupItems) {
+        super(name);
+        this.backupItems = backupItems;
+    }
+
+    @Override
+    public void run() {
+        QueueContainer container = getContainer();
+        Map<Long, QueueItem> backupMap = container.getBackupMap();
+        backupMap.clear();
```

Comment:
This may be a minor issue - if the policy returns `null`, we keep the container instead of destroying it as well. We may think about this as an optimisation though.
Also, maybe we can call `destroy` here to destroy the transaction data as well.